app/services/claimToAdjustPoa/SubmitPoaHelper.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.claimToAdjustPoa
+
+import auth.MtdItUser
+import config.featureswitch.FeatureSwitching
+import config.{AgentItvcErrorHandler, ItvcErrorHandler}
+import controllers.agent.predicates.ClientConfirmedController
+import controllers.routes.HomeController
+import models.admin.AdjustPaymentsOnAccount
+import models.claimToAdjustPoa.{PaymentOnAccountViewModel, PoAAmendmentData}
+import models.core.Nino
+import play.api.Logger
+import play.api.mvc.Result
+import services.{ClaimToAdjustPoaCalculationService, ClaimToAdjustService, PaymentOnAccountSessionService}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait SubmitPoaHelper extends ClientConfirmedController with FeatureSwitching {
```

Comment:
:wrench: - RecalculatePoaHelper or similar, I am not sure if "Submit" is something related to this domain, we need to be domain specific so to easy read this code in the future;